sys/phydat/phydat_str.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_sensif
+ * @{
+ *
+ * @file
+ * @brief       Generic sensor/actuator data handling
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+
+#include "phydat.h"
+
+void phydat_dump(phydat_t *data, int dim)
+{
+    if (data == NULL || dim < 0 || dim > PHYDAT_DIM) {
+        puts("Unable to display data object");
+        return;
+    }
+    printf("Data:");
+    for (int i = 0; i < dim; i++) {
+        printf("\t[%i] - %i%s%s\n", i, data->val[i],
+               phydat_scale_to_str(data->scale),
+               phydat_unit_to_str(data->unit));
+    }
+}
+
+const char *phydat_unit_to_str(uint8_t unit)
+{
+    switch (unit) {
+        case UNIT_DEG_C:    return "°C";
+        case UNIT_DEG_F:    return "°F";
+        case UNIT_DEG_K:    return "°K";
+        case UNIT_M:        return "m";
+        case UNIT_G:        return "g";
+        case UNIT_DPS:      return "dps";
+        case UNIT_GR:       return "G";
+        case UNIT_A:        return "A";
+        case UNIT_V:        return "V";
+        case UNIT_GS:       return "Gs";
+        case UNIT_BAR:      return "Bar";
+        case UNIT_PA:       return "Pa";
+        case UNIT_CD:       return "cd";
+        default:            return "";
+    }
+}
+
+const char *phydat_scale_to_str(int8_t scale)
+{
+    switch (scale) {
+        case 0:     return "";
+        case -3:    return "m";
+        case -6:    return "u";
+        case -9:    return "n";
+        case -12:   return "p";
+        case -15:   return "f";
+        case 1:     return "d";
+        case 2:     return "c";
+        case 3:     return "k";
+        case 6:     return "M";
+        case 9:     return "G";
+        case 12:    return "T";
+        default:    return "*10^X";     /* TODO: how to output this here */
```

Comment:
Is it though? I can imagine using this to be send directly over CoAP as output data.